requests/maniphest_search.go

```diff
@@ -0,0 +1,116 @@
+package requests
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/uber/gonduit/entities"
+)
+
+// ManiphestSearchRequest represents a request to maniphest.search API method.
+type ManiphestSearchRequest struct {
+	// QueryKey is builtin or saved query to use. It is optional and sets initial constraints.
+	QueryKey    string             `json:"queryKey,omitempty"`
+	// Constraints contains additional filters for results. Applied on top of query if provided.
+	Constraints *SearchConstraints `json:"constraints,omitempty"`
+	// Attachments specified what additional data should be returned with each result.
+	Attachments *SearchAttachments `json:"attachments,omitempty"`
+
+	*entities.Cursor
+	Request
+}
+
+// SearchAttachments contains fields that specify what additional data should be returned with sesarch results.
+type SearchAttachments struct {
+	// Subscribers if true instructs server to return subscribers list for each task.
+	Subscribers bool `json:"subscribers,omitempty"`
+	// Columns requests to get the workboard columns where an object appears.
+	Columns     bool `json:"columns,omitempty"`
+	// Projects requests to get information about projects.
+	Projects    bool `json:"projects,omitempty"`
+}
+
+// ManiphestRequestSearchOrder describers how results should be ordered.
+type ManiphestRequestSearchOrder struct {
+	// Builtin is the name of predefined order to use.
+	Builtin string
+	// Order is list of columns to use for sorting, e.g. ["color", "-name", "id"],
+	Order   []string
+}
```

Comment:
There doesn't seem to be precedent in this repository for search ordering so
we have opportunity to set one.

From the API docs, this is either a list of fixed strings, or a list of
columns from a per-defined list with the following constraints.

    each column may only be specified once;
    each column may be prefixed with - to invert the order;
    the last column must be a unique column, usually id; and
    no column other than the last may be unique.

This isn't the best way to represent this information in the user API.

We can discuss the right method of representing the custom ordering API if you
have need of it. Otherwise, for now you can probably support only the fixed
list of strings.

Reply:
@abhinav  I don't need custom ordering for my use case. But I feel like removing it and only leaving Builtin field will result in partial support of this feature. 
What is your proposal?